corehq/apps/case_importer/tracking/dbaccessors.py

```diff
@@ -29,7 +31,11 @@ def get_case_upload_record_count(domain, user):
 
 def get_case_ids_for_case_upload(case_upload):
     for form_record in case_upload.form_records.order_by('pk').all():
-        form = XFormInstance.objects.get_form(form_record.form_id, case_upload.domain)
+        if toggles.DOMAIN_PERMISSIONS_MIRROR.enabled(case_upload.domain) \
+                or EnterprisePermissions.get_by_domain(case_upload.domain).is_enabled:
```

Comment:
The absolute strictest thing to do would be to verify both that the permissions object is enabled and also that its `source_domain` is `case_upload.domain` (only the source domain has access to other domains).